swagger.json

```diff
@@ -1170,6 +1170,10 @@
           "comment": {
             "description": "Optional URL-escaped reason why sign-off is PASS or FAIL",
             "type": "string"
+          },
+          "release": {
+            "description": "Optional boolean for release status",
```

Comment:
I don't understand what this means. The status of the release is true or false?

Reply:
I think it's either "there will be a release" or "there will not be a release" or NULL

Reply:
Okay, I think using the word 'status' to describe future plans is confusing, that could easily be interpreted as 'has been released' or 'has not been released'

Also, is there a meaningful difference between 'there will not be a release' and null, or is there just no mechanism to disallow null?

Reply:
What will happen when an Analysis Review signoff is set to release:true? Or a Release Approval has release as null?

This field seems to me very overloaded and it's just being introduced. All of these qc step types are the same 'signoffDao'? Should all of these be split out into different classes so that they have more concretely described behaviours in these cases?

Reply:
Dimsum is going to have QC statuses with labels that are meant for humans to understand. These statuses will tell us two things:

1. QC status: pass or fail
2. Release status: release/released, or don't/not released

e.g. Release Approval status "Failed/Proceed" is qc_passed == false, release == true, and would represent something like a failed clinical report - it's a failure, but we want to proceed to release

We could make this more complicated by defining the statuses fully in the Nabu database, but there's no reason to do that as

1. they are rarely, if ever, going to change, and
2. they are only relevant within Dimsum

This data and API is intended for use by Dimsum and only Dimsum

Reply:
Aren't QC Statuses controlled by a clinical SOP? Shouldn't the system protect against non-SOP states? Dimsum is the only system that writes to nabu for now, but we don't know what will happen in the future. "Dimsum will always do the right thing" is also assuming dimsum will never have a bug in it. I think it would be better to have nabu reject bad states and display an error in dimsum than let dimsum write invalid data to nabu.